### DISCUSSION TITLE
HC_API_URL env variable doesn't work as expected

### DISCUSSION THREAD

**Discussion by rwjack** [bdd/runitor#27] [General]
Method 1 - Works:
```
runitor -api-url="https://example.com/ping" -uuid 123-abc-456-def -- echo test
```
Method 2 - Works:
```
HC_API_URL="https://example.com/ping" runitor -uuid 123-abc-456-def -- echo test
```
Method 3 - Doesn't work:
```
HC_API_URL="https://example.com/ping"
runitor -uuid 123-abc-456-def -- echo test
runitor -uuid 789-ghi-012-jkl -- echo test2
```
Tested all sorts of combinations, can't get cron to execute method 3, which is the cleanest one for having multiple jobs, but only if I could set the HC_API_URL variable.

**Comment by bdd** [comment]
Method 3 would only work if it was a shell script, executed by the cronjob.
If you're adding these as separate cron job entities, they execute separately, without state.

If these jobs needs to execute in different schedules, but not serially, then you have to prefix your cron entries either with the environment variable like in Method 2 or pass the `-api-url` parameter in each.

**Reply by rwjack** [reply]
I did try to define `HC_API_URL="https://example.com/ping"` in cron's environment by writing it down directly in the `crontab -e` file, and was hoping we could get it exported globally so cron would look cleaner.

And by doing:
```
* * * * * env > /root/test
```
Indeed `HC_API_URL` was defined as an env variable, but it did not seem to work. Might be just me lacking the vital understanding of either cron's, or the general environment.

But a simple test:
```
HC_API_URL="https://example.com/ping"
runitor -uuid 123-abc-456-def -- echo test
```
Showed that it does not work serially as I expected, which relates to my issue - why not have runitor read the global variable, instead of having to write it down serially like:
`HC_API_URL="https://example.com/ping" runitor -uuid 123-abc-456-def -- echo test`

Because if this is the case, then I don't see the point of these 2 different commands doing the same thing:
`HC_API_URL="https://example.com/ping" runitor -uuid 123-abc-456-def -- echo test`
or
`runitor -api-url="https://example.com/ping" -uuid 123-abc-456-def -- echo test`

**Reply by rwjack** [reply]
Quick update,
setting
`export HC_API_URL="https://example.com/ping"` in a bash script works, therefore I have also tried the export in `/etc/environment`, which seems to be working for cron, but also for all users. I bet there has to be a way to set it up only for root.

**Comment by cuu508** [comment]
The crontab file format supports environment variables. From `man 5 crontab`:

> An  active  line  in a crontab will be either an environment setting or a cron command.  The crontab file is parsed from top to bottom, so any environment settings will affect only the cron commands
> below them in the file.  An environment setting is of the form,
> 
>    name = value
> 
> where the spaces around the equal-sign (=) are optional, and any subsequent non-leading spaces in value will be part of the value assigned to name.  The value string may be placed in quotes  (single
> or double, but matching) to preserve leading or trailing blanks.  To define an empty variable, quotes must be used.

As a quick experiment, I set up my user's crontab (using `crontab -e`) like so:

```
HC_PING_KEY = redacted-ping-key
FOO = barry
* * * * * /usr/local/bin/runitor -slug test-slug -- echo hey $FOO
```

And it worked – I started seeing pings with body "hey barry" in Healthchecks dashboard. This means that

* runitor had access to `HC_PING_KEY` env var – otherwise it would not be able to submit the ping
* the payload command had access to `FOO` env var – otherwise I would not see "barry" in the request payload

**Comment by BeyondVertical** [comment]
Just an FYI: I just updated to v1.1.0 and before I had the URL of my custom Healthchecks instance without a "/" at the end. Which worked with v1.0.0.
But after updating the pings did not work. I then added the "/" at the end of the URL and now it works again with v1.1.0. :)

**Reply by bdd** [reply]
#75
